File: src/main/java/com/beat/domain/booking/application/dto/TicketRetrieveResponse.java
package com.beat.domain.booking.application.dto;

import java.util.List;

public record TicketRetrieveResponse(
	String performanceTitle,
	int totalScheduleCount,
	List<TicketDetail> bookingList
) {
	public static TicketRetrieveResponse of(
		String performanceTitle,
		int totalScheduleCount,
		List<TicketDetail> bookingList) {
		return new TicketRetrieveResponse(performanceTitle, totalScheduleCount, bookingList);
	}
}

File: apis/src/main/kotlin/com/beat/apis/ticket/api/response/TicketRetrieveResponse.kt
package com.beat.apis.ticket.api.response

import com.beat.application.frontoffice.ticket.maker.query.TicketRetrieveResult

@ConsistentCopyVisibility
data class TicketRetrieveResponse private constructor(
    val performanceTitle: String?,
    val performanceTeamName: String?,
    val totalScheduleCount: Int,
    val totalPerformanceTicketCount: Int,
    val totalPerformanceSoldTicketCount: Int,
    val bookingList: List<TicketDetail>,
) {
    companion object {
        fun from(result: TicketRetrieveResult): TicketRetrieveResponse = TicketRetrieveResponse(
            performanceTitle = result.performanceTitle,
            performanceTeamName = result.performanceTeamName,
            totalScheduleCount = result.totalScheduleCount,
            totalPerformanceTicketCount = result.totalPerformanceTicketCount,
            totalPerformanceSoldTicketCount = result.totalPerformanceSoldTicketCount,
            bookingList = result.bookingList.map(TicketDetail::from),
        )
    }
}
